core/filtermaps/map_renderer.go

```diff
@@ -257,11 +256,14 @@ func (f *FilterMaps) loadHeadSnapshot() error {
 
 // makeSnapshot creates a snapshot of the current state of the rendered map.
 func (r *mapRenderer) makeSnapshot() {
-	r.f.renderSnapshots.Add(r.iterator.blockNumber, &renderedMap{
+	if r.iterator.blockNumber != r.currentMap.lastBlock {
+		panic("iterator state inconsistent with last block")
```

Comment:
Probably discard the snapshot instead of panic here?

Reply:
I'd prefer to panic here because it is something that absolutely should not and will not happen; `currentMap` and `iterator` are both needed here and should be in a consistent state which is currently trivially ensured by the caller. If a future change ever breaks this then I think the louder and harder it breaks, the better.